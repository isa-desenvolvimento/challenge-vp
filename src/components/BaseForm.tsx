import { Formik, Form } from 'formik'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const BaseForm = ({ initialValues, baseForm, onSubmit }: any) => {
  const renderForm = () => {
    return (
      <Formik
        onSubmit={(data, action) => onSubmit(data, action)}
        initialValues={initialValues}
      >
        {() => <Form>{baseForm}</Form>}
      </Formik>
    )
  }

  return <>{renderForm()}</>
}
