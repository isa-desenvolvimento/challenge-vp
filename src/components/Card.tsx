import {
  StyleCardContainer,
  StyleCardContent,
  StyleCardFooter,
  StyleCardHeader,
} from '@/styles/Card'
import { CardProps } from '@/types/global'

export const Card = ({
  header,
  content,
  footer,
  primary,
  height,
}: CardProps) => {
  const renderHeader = () =>
    header && <StyleCardHeader primary={primary}>{header}</StyleCardHeader>
  const renderContent = () =>
    content && (
      <StyleCardContent primary={primary} height={height}>
        {content}
      </StyleCardContent>
    )
  const renderFooter = () =>
    footer && <StyleCardFooter primary={primary}>{footer}</StyleCardFooter>

  return (
    <StyleCardContainer role="card">
      {renderHeader()}
      {renderContent()}
      {renderFooter()}
    </StyleCardContainer>
  )
}
