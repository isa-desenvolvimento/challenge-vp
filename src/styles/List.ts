import { ListProps } from '@/types/global'
import styled from 'styled-components'

export const StyleListContainer = styled.div``

export const StyAmount = styled.span`
  margin-top: 1rem;
  font-weight: 500;
  color: ${({ theme }) => theme.colors.active.contrast};
`

export const StyItemList = styled.div.attrs((props: ListProps) => ({
  withBorder: props.withBorder,
  items: props.items,
}))<ListProps>`
  position: relative;
  width: 100%;
  margin-left: 0px;
  margin-right: 0px;
  display: grid;
  grid-auto-columns: 40% auto;
  padding: 1rem;
  border-bottom: ${({ theme, withBorder }) =>
    withBorder && `1px solid ${theme.pallet.gainsboro}`};
  :hover {
    background-color: ${({ theme }) => theme.colors.background};
  }

  display: flex;
  justify-content: space-between;
  align-items: flex-end;
`
