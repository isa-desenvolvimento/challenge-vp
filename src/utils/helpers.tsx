import { render } from '@testing-library/react'
import GlobalContext from '../context'

export const renderWithProvider = (children: React.ReactNode) => {
  render(<GlobalContext>{children}</GlobalContext>)
}
