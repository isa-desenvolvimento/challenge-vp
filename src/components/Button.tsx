import { Icon } from '@/components/Icon'
import { StyleButton } from '@/styles/Button'
import { ButtonProps } from '@/types/global'

export const Button = ({
  iconProps,
  primary,
  onClick,
  children,
  size,
  type,
}: ButtonProps) => {
  const renderIcon = () => iconProps && <Icon faIcon={iconProps}></Icon>

  return (
    <StyleButton
      role="button"
      arial-label="button"
      primary={primary}
      size={size}
      type={type}
      onClick={onClick}
    >
      {renderIcon()}
      {children}
    </StyleButton>
  )
}
