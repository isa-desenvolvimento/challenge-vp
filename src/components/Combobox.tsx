import { ComboboxProps } from '@/types/global'

import { DropdownWrapper, StyledSelect, StyledOption } from '@/styles/Combobox'
import { useField } from 'formik'

export const Combobox = ({
  role,
  options,
  placeholder,
  primary,
  onChange,
  name,
  isFormik,
}: ComboboxProps) => {
  const [field, meta] = useField(name)

  const renderSelection = () => {
    if (isFormik) {
      return (
        <StyledSelect primary={primary} {...field} {...meta} role={role}>
          <StyledOption selected value="" name={`${name}.defaultValue`}>
            {placeholder}
          </StyledOption>
          {renderOptions()}
        </StyledSelect>
      )
    }

    return (
      <StyledSelect onChange={onChange} primary={primary}>
        <StyledOption selected value="" name={`${name}.defaultValue`}>
          {placeholder}
        </StyledOption>
        {renderOptions()}
      </StyledSelect>
    )
  }

  const renderOptions = () => {
    return (
      options?.length &&
      options.map(({ value }) => {
        return (
          <StyledOption name={`${name}.${value}`} key={value} value={value}>
            {value}
          </StyledOption>
        )
      })
    )
  }

  return <DropdownWrapper role="combobox">{renderSelection()}</DropdownWrapper>
}
