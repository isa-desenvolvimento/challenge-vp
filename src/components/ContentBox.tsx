import { StyleContainer } from '@/styles/ContentBox'
import { ChildrenProps } from '@/types/global'

export const ContentBox = ({ children }: ChildrenProps) => {
  return <StyleContainer role="contentBox"> {children}</StyleContainer>
}
