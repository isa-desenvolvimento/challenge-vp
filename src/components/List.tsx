import { StyAmount, StyItemList, StyleListContainer } from '@/styles/List'
import { TagState } from '@/components/TagState'
import { TitleDescription } from './TitleDescription'
import { ItemProps, ListProps } from '@/types/global'
import { formatMoeda } from '@/utils/formatNumer'
import { formatDate } from '@/utils/formatDate'

export const List = ({ withBorder, withTag, items }: ListProps) => {
  const renderItems = () => {
    return items.map((item: ItemProps) => (
      <StyItemList withBorder={withBorder} key={item.id} role="list">
        <TitleDescription
          title={item.title}
          description={item.description}
          date={item.date ? formatDate(item.date) : ''}
        />
        {withTag && <TagState status={item.status}>{item.status}</TagState>}
        <StyAmount>{formatMoeda(item.amount)} </StyAmount>
      </StyItemList>
    ))
  }
  return <StyleListContainer>{renderItems()}</StyleListContainer>
}
