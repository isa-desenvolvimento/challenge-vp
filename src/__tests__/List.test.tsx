import { renderWithProvider } from '../utils/helpers'
import { screen } from '@testing-library/react'

import { List } from '@/components/List'

const items = [
  {
    id: 1,
    title: 'List',
    description: 'List',
    date: '2016-02-12',
    amount: 2.0,
    status: 'done',
  },
]

describe('<List />', () => {
  beforeEach(() =>
    renderWithProvider(<List withBorder withTag items={items} />)
  )

  it('should render text List', () => {
    expect(screen.getByRole(/list/i).textContent).toContain('List')
  })

  it('should render snapshot', () => {
    expect(screen.getByRole(/list/i)).toMatchSnapshot()
  })
})
