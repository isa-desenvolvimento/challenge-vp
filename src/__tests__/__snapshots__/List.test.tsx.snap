// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render snapshot 1`] = `
.c5 {
  margin-top: 1rem;
  font-weight: 500;
  color: #36a4a8;
}

.c0 {
  position: relative;
  width: 100%;
  margin-left: 0px;
  margin-right: 0px;
  display: grid;
  grid-auto-columns: 40% auto;
  padding: 1rem;
  border-bottom: 1px solid #E3E3E3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c0:hover {
  background-color: #eff0f6;
}

.c4 {
  position: absolute;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  padding: 0rem 2rem;
  color: #ffff;
  background: #6CD188;
  left: 50%;
  -webkit-transform: translate(-50%);
  -ms-transform: translate(-50%);
  transform: translate(-50%);
  width: 8rem;
  top: 0;
  box-shadow: 0px 2px 5px rgb(0 0 0 / 12%),0 0px 10px rgb(0 0 0 / 8%);
}

.c1 {
  text-align: justify;
  margin: 1rem;
}

.c2 {
  font-weight: bold;
  color: #60647C;
}

.c3 {
  color: #ddd;
}

<div
  class="c0"
  role="list"
>
  <div
    class="c1"
  >
    <h5
      class="c2"
    >
      List
    </h5>
    <h5
      class="c3"
    >
      List
    </h5>
    <h5
      class="c3"
    >
      2016-02-12
    </h5>
  </div>
  <div
    class="c4"
    role="tag-status"
  >
    done
  </div>
  <span
    class="c5"
  >
    R$Â 2,00
     
  </span>
</div>
`;
